wordpress-mobile/WordPress-iOS
# Blogging Reminders - Time Selection

Fixes #NA

To test:
use an account with more than one site
- Before running this branch, build and run another branch (e.g. `develop`) and set some Blogging Reminders to some of the sites (go to Site Settings -> Blogging Reminders, select some days and tap on "Notify Me")

Repeat the following both on a site with pre-existing reminders and on one that did not have reminders
- Build/run this branch then go to Site Settings -> Blogging Reminders
- if you selected a site with previously set reminders, make sure the reminders still show up.
- Select one or more days: this should make the time selection button appear. If no time was selected before, the default time will be 10:00 AM
- tap on the time selection button, you should see a time picker (see #17024 for details on the UI)
- select a time when to schedule a notification (tip: select a time in the near future, e.g. the next minute or two in the same day of the week as today, to avoid waiting too long for the reminder to show up)
- Make sure the title of the time picker screen updates with the selected time
- Tap "Back" and make sure the time in the selection button updates with the time you selected
- Tap "Notify Me" (or "Update, if you changed an existing schedule) and then "All Set" on the epilogue screen (alternatively, just dismiss the epilogue screen)
- Make sure the console logs the following tracking event:

`🔵 Tracked: blogging_reminders_scheduled <blog_type: wpcom, days_of_week_count: 1, selected_time: 12:49>`

- Wait for the notification to show up at the time you selected, and make sure the reminder shows up
- Optionally, go in the tracks live view, wait (about 5 mins) for the above event (`wpios_blogging_reminders_scheduled`) to show up and make sure the `selected_time` property is consistent with what you've set (the time will be standard 24h `HH:mm`)

## Regression Notes
1. Potential unintended areas of impact
This change refactors a few parts of Blogging Reminders scheduler and changes the format that was previously used for reminders, so it's important to ensure backward compatibility

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Install this branch on top of an old one that used the previous scheduling format; tested that notifications are actually arriving at the desired time

3. What automated tests I added (or what prevented me from doing so)
none

PR submission checklist:
